nrf52-hal-common/src/twim.rs

```diff
@@ -110,151 +112,151 @@ impl<T> Twim<T> where T: TwimExt {
 
     /// Write to an I2C slave
     ///
-    /// The buffer must have a length of at most 255 bytes on the nRF52832
-    /// and at most 65535 bytes on the nRF52840.
     pub fn write(&mut self,
         address: u8,
         buffer:  &[u8],
     )
         -> Result<(), Error>
     {
-
-        if buffer.len() > EASY_DMA_SIZE {
-            return Err(Error::TxBufferTooLong);
+        if SRAM_LOWER <= ( buffer.as_ptr() as usize ) && (buffer.as_ptr() as usize) < SRAM_UPPER {
```

Comment:
As pointed out in https://github.com/nrf-rs/nrf52-hal/pull/50/files/968be9f40c1b04f3b7715a7f947a20707f8580b4#r262476494 this should be fixed.